source/common/network/listen_socket_impl.cc

```diff
@@ -15,6 +15,12 @@ void ListenSocketImpl::doBind() {
     throw EnvoyException(
         fmt::format("cannot bind '{}': {}", local_address_->asString(), strerror(errno)));
   }
+  if (local_address_->type() == Address::Type::Ip && local_address_->ip() != nullptr &&
```

Comment:
nit: local_address_->ip() != nullptr is guaranteed if local_address_->type() == Address::Type::Ip